docs/codeql/codeql-language-guides/customizing-library-models-for-python.rst

```diff
@@ -0,0 +1,449 @@
+- **kind**: Kind of summary to add. Can be **taint** for taint-propagating flow, or **value** for value-preserving flow.
+
+Example:
+
+.. code-block:: yaml
+
+  extensions:
+    - addsTo:
+        pack: codeql/python-all
+        extensible: summaryModel
+      data:
+        - [
+            "builtins",
+            "Member[reversed]",
+            "Argument[0]",
+            "ReturnValue",
+            "taint",
+          ]
+
+typeModel(type1, type2, path)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A description of how to reach **type1** from **type2**.
+If this is the only way to reach **type1**, for instance if **type1** is a name we made up to represent the inner workings of a library, we think of this as a definition of **type1**.
+In the context of instances, this describes how to obtain an instance of **type1** from an instance of **type2**.
+
+- **type1**: Name of the type to reach.
+- **type2**: Name of the type from which to evaluate **path**.
+- **path**: Access path leading from **type2** to **type1**.
+
+Example:
+
+.. code-block:: yaml
+
+  extensions:
+  - addsTo:
+      pack: codeql/python-all
+      extensible: typeModel
+    data:
+      - [
+          "flask.Response",
+          "flask",
+          "Member[jsonify].ReturnValue",
+        ]
+
+Types
+-----
+
+A type is a string that identifies a set of values.
+In each of the extensible predicates mentioned in previous section, the first column is always the name of a type.
+A type can be defined by adding **typeModel** tuples for that type. Additionally, the following built-in types are available:
+
+- The name of a package matches imports of that package. For example, the type **django** matches the expression **import django**.
+- The type **builtins** identifies the builtins package. In Python, all built-in values are found in this package, so they can be identified using this type.
+- A dotted path ending in a class name identifies instances of that class. If the suffix **!** is added, the type refers to the class itself.
+
+Access paths
+------------
+
+The **path**, **input**, and **output** columns consist of a **.**-separated list of components, which is evaluated from left to right, with each step selecting a new set of values derived from the previous set of values.
+
+The following components are supported:
+
+- **Argument[**\ `number`\ **]** selects the argument at the given index.
+- **Argument[this]** selects the receiver of a method call.
+- **Parameter[**\ `number`\ **]** selects the parameter at the given index.
+- **Parameter[this]** selects the **this** parameter of a function.
+- **ReturnValue** selects the return value of a function or call.
+- **Member[**\ `name`\ **]** selects the function/method/class/value with the given name.
+- **Instance** selects instances of a class, including instances of its subclasses.
+- **Attribut[**\ `name`\ **]** selects the attribute with the given name.
```

Comment:
```suggestion
- **Attribute[**\ `name`\ **]** selects the attribute with the given name.
```

?

Reply:
Thanks for catching this! I have tried to use copy and paste to get these right (copying from the implementation, so it is the actual string we recognise) but somehow it has gone wrong here...